libs/sdk-ui-kit/src/Dialog/ConversionTouchPointDialogBase.tsx

```diff
@@ -0,0 +1,130 @@
+// (C) 2021 GoodData Corporation
+import React, { useEffect, useState } from "react";
+import { useIntl } from "react-intl";
+import { useHubspotForm } from "@aaronhayes/react-use-hubspot-form";
+import { LoadingComponent } from "@gooddata/sdk-ui";
+
+import { Button } from "../Button";
+import { HubSpotFormLocale } from "@aaronhayes/react-use-hubspot-form/build/main/hubspot-form-locale.enum";
+
+declare global {
+    interface Window {
+        jQuery: object;
+        hbspt: undefined | { isSuccessMessageShow: boolean };
+    }
+}
+
+/**
+ * @internal
+ */
+export interface HubspotFormField {
+    name: string;
+    value: string | any;
+    click: () => void;
+}
+
+/**
+ * @public
+ */
+export interface IConversionTouchPointDialogBaseProps {
+    selectedValue?: string;
+    userEmail: string;
+    clientId: string | number;
+    dialogTitle: string;
+    cancelButtonText?: string;
+    showCancelButton?: boolean;
+    onClose: () => void;
+    onFormSubmitted: () => void;
+}
+
+/**
+ * @internal
+ */
+export const ConversionTouchPointDialogBase: React.FC<IConversionTouchPointDialogBaseProps> = (props) => {
+    const HUBSPOT_EMAIL_FIELD_NAME = "email";
+    const HUBSPOT_GACID_FIELD_NAME = "gacid";
+    const HUBSPOT_PORTAL_ID = "8934247";
+    const HUBSPOT_FORM_ID = "94ade70f-4052-432d-9453-5cbf26981926";
+    const {
+        selectedValue,
+        userEmail,
+        clientId,
+        cancelButtonText,
+        dialogTitle,
+        showCancelButton,
+        onClose,
+        onFormSubmitted,
+    } = props;
+
+    const intl = useIntl();
+    const [isFormReady, setIsFormReady] = useState(false);
+
+    useEffect(() => {
+        // the Hubspot form requires jQuery object, we need to create a fake jQuery object to bypass this.
+        window.jQuery =
+            window.jQuery ||
+            function (nodeOrSelector: string | object) {
+                if (typeof nodeOrSelector == "string") {
+                    return document.querySelector(nodeOrSelector);
+                }
+                return nodeOrSelector;
+            };
+        if (window.hbspt?.isSuccessMessageShow) {
+            window.hbspt.isSuccessMessageShow = false;
+        }
+    }, []);
+
+    const onHubspotFormSubmitted = () => {
+        if (!window.hbspt.isSuccessMessageShow) {
+            onFormSubmitted();
+            onClose();
+            window.hbspt.isSuccessMessageShow = true;
+        }
+    };
+
+    const { formCreated } = useHubspotForm({
+        portalId: HUBSPOT_PORTAL_ID,
+        formId: HUBSPOT_FORM_ID,
+        target: "#conversion-touch-point-hubspot",
```

Comment:
As mentioned in the other comment, can hard-coding `target` as `#conversion-touch-point-hubspot` here cause any issue if other customers also use `ConversionTouchPointDialogBase` in their apps?

Reply:
Updated. Thanks!